### PR TITLE
Fixes #9713: restore WIN32 service support

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by dmitri-d** [comment]
This is based on https://github.com/theforeman/smart-proxy/pull/265. The main change is that windows service-specific code has been moved into a dedicated file in bin/ directory.

**Comment by mchesler** [comment]
:+1: Need this badly!

**Comment by claer** [comment]
It is possible to update the extra/register-service.rb script at the same time?

**Comment by dmitri-d** [comment]
@claer: done and done.

**Comment by mchesler** [comment]
Per the 1.8.0 documentation, Ruby 1.9.3p551 is now the recommended version.  Running this with 1.9.3 results in an exception because the PLATFORM variable has been deprecated.  RUBY_PLATFORM should be used instead.

**Comment by claer** [comment]
@mchesler I just hit the same problem (and I'm the one who updated the documentation in favor of 1.9.3, shame on me)

@witlessbird I've done the modifications by hand and still no luck, I'll report later in the evening.

**Comment by dmitri-d** [comment]
@claer, @mchesler: if you have a backtrace, could you post it here?

**Comment by dmitri-d** [comment]
or you can post it in the redmine ticket.

**Comment by mchesler** [comment]
@witlessbird - here you go:

```
C:\smart-proxy-1.8.0>ruby extra\register-service.rb
extra/register-service.rb:2:in `<main>': uninitialized constant PLATFORM (NameError)
```

**Comment by dmitri-d** [comment]
[test]

**Comment by dmitri-d** [comment]
@mchesler: should do the trick...

**Comment by dmitri-d** [comment]
@claer: removed extraneous "--service" when registering windows service.

**Comment by dmitri-d** [comment]
@claer, @mchesler: is this good to merge, or are there still issues remaining?

File: bin/smart-proxy-win-service [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+raise "win-service executable can only be used as a service in windows environment" unless RUBY_PLATFORM =~ /mingw/
+
+require 'smart_proxy'
+require 'win32/daemon'
+include Win32
+
+begin
+  # Logfile must be absolute on windows
+  logfile = "../../tmp/proxy.log"
```
Comment (mchesler) [inline bin/smart-proxy-win-service:12]: @witlessbird seems wrong to have a comment about needed absolute path on windows and then setting a relative path.

**Comment by dmitri-d** [comment]
@mchesler: removed the comment you pointed out.

**Comment by mchesler** [comment]
:+1:

**Comment by dmitri-d** [comment]
@domcleal, @lzap: could one of you take another look and merge this PR? The changes are windows-specific and have no effect on other parts of smart-proxy.

**Comment by domcleal** [comment]
Thanks @witlessbird, @mchesler and @claer!  Merged as a02e0f1.

**Comment by dmitri-d** [comment]
And thanks to @mmatuska who's the original author of this PR!

**Pull request closed without merging**